world-server/src/world/chat/ChatHandlers.ts

```diff
@@ -0,0 +1,46 @@
+import { Messages, Router, Logger } from 'world-core';
+import { BroadcastMessage } from '../../network';
+import { ChatMessage } from './ChatMessages';
+import { Player } from '../player';
+
+export class ChatHandler implements Messages.MsgHandler {
+    private players: Player[];
+
+    constructor(players: Player[]) {
+        this.players = players;
+    }
+
+    public getTypes(): string[] {
+        return ['chat.say'];
+    }
+
+    public validate(msg: Messages.Message): boolean {
+        let message = msg as ChatMessage;
+        if (message == null || message == undefined) {
+            Logger.warn(`Received player message does not extend ChatMessage`);
```

Comment:
Technically incorrect comment. Casts in Typescript are only for developers. At the end, JS is executed and no runtime type checks are present.

Reply:
I didn't write this but I guess that this could be triggered if a client sends a `chat.say` message but uses message contents for another message type. I think we could keep this?

Reply:
Asked @davidmohar about this, says it's impossible, removing the logger warning.